front/app/components/Form/Components/Layouts/CLPageLayout.tsx

```diff
@@ -0,0 +1,160 @@
+import React, { memo, useState, useEffect, useContext } from 'react';
+import { LayoutProps, RankedTester, rankWith } from '@jsonforms/core';
+import { JsonFormsDispatch, withJsonFormsLayoutProps } from '@jsonforms/react';
+import styled, { useTheme } from 'styled-components';
+import Ajv from 'ajv';
+
+// Components
+import {
+  Box,
+  Title,
+  useBreakpoint,
+  media,
+} from '@citizenlab/cl2-component-library';
+import { FormSection } from 'components/UI/FormComponents';
+import Button from 'components/UI/Button';
+
+// Context
+import { FormContext } from 'components/Form/contexts';
+
+// i18n
+import { FormattedMessage } from 'utils/cl-intl';
+import messages from '../../messages';
+
+// Utils
+import {
+  getSanitizedFormData,
+  getPageSchema,
+  PageCategorization,
+  isPageCategorization,
+} from 'components/Form/Components/Layouts/utils';
+
+const StyledFormSection = styled(FormSection)`
+  max-width: 100%;
+  width: 100%;
+
+  ${media.phone`
+    padding-left: 18px;
+    padding-right: 18px;
+  `}
+`;
+
+const customAjv = new Ajv({ useDefaults: 'empty', removeAdditional: true });
+
+const CLPageLayout = memo(
+  ({
+    schema,
+    uischema,
+    path,
+    renderers,
+    cells,
+    enabled,
+    data,
+  }: LayoutProps) => {
+    const { setShowSubmitButton, onSubmit, setShowAllErrors, formSubmitText } =
+      useContext(FormContext);
+    const [currentStep, setCurrentStep] = useState<number>(0);
+    const uiCategories = (uischema as PageCategorization).elements;
+    const theme: any = useTheme();
+    const isSmallerThanXlPhone = useBreakpoint('phone');
+    const submitText = formSubmitText || messages.submit;
```

Comment:
Just wondering - in which case is there a different "formSubmitText"?

Reply:
We want to abstract survey data out of the implementation as much as possible. In this case, the PageLayout should handle any pages. Our submit text is "Submit survey". We don't want that in here so we provide a way for it to be passed in and default to Submit.